sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeActionsOperation.cs

```diff
@@ -127,7 +134,19 @@ public class AnalyzeActionsOperation : PageableOperation<AnalyzeActionsResult>,
         /// <param name="client">The client used to check for completion.</param>
         public AnalyzeActionsOperation(string operationId, TextAnalyticsClient client)
         {
-            // TODO: Add argument validation here.
+            try
+            {
+                string plainOperationId = ClientCommon.DecodeOperationId(operationId);
+                OperationIdInformation idInformation = JsonSerializer.Deserialize<OperationIdInformation>(plainOperationId);
```

Comment:
Can we put all this logic for decoding inside the `OperationIdInformation` class?  It feels less likely to get out of sync if it's all contained in one location.

How does this version?  I'm thinking about scenarios where we add some future property and here I think it'd just be ignored, but maybe we should fail instead if we're given something we don't fully understand?

Reply:
Moved all logic into `Decode` and `Encode` methods in `OperationIdInformation`. I left the try/catch inside the LRO constructors.

Reply:
> How does this version? I'm thinking about scenarios where we add some future property and here I think it'd just be ignored, but maybe we should fail instead if we're given something we don't fully understand?

This would only happen if a user tried to pass the ID of a LRO started by a newer SDK version. Is this a common scenario? Even if we tried to pass the new properties somehow, the behavior would be limited by the service version.

One thing we could do is using the AdditionalInfo dictionary, as you suggested, and we throw if we find any non-null property defined (non-supported behavior for this version). Otherwise, LRO works normally.

What do you think?